src/openai/_base_client.py

```diff
@@ -1549,7 +1549,7 @@ async def _request(
         if self._platform is None:
             # `get_platform` can make blocking IO calls so we
             # execute it earlier while we are in an async context
-            self._platform = await asyncify(get_platform)()
+            self._platform = await asyncio.to_thread(get_platform)
```

Comment:
We can't switch to just using `asyncio.to_thread()` itself as that was only added in Python 3.9 but we support Python 3.8+

Thankfully it looks like it mostly just delegates to [`loop.run_in_executor()`](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor) under the hood which is available in older versions.

Can you instead change the `asyncify` function to use `asyncio.to_thread()` on 3.9 and fallback to a copied version of the [source](https://github.com/python/cpython/blob/a1c57bcfd2bcbc55ff858407e09c1d8d8cee44e6/Lib/asyncio/threads.py#L12)? or just always use the copied version? I don't feel strongly either way.

You can remove the other `cancellable` and `limiter` args from `asyncify` as I don't think they make sense anymore.

Reply:
@RobertCraigie, since I don't have access to @jeongsu-an 's repository, I forked the repo and put up a PR for the changes you requested in #1853.  Happy to iterate there if you have suggestions or changes you'd like.